### ISSUE TITLE
Only `exec.go` gets generated

### ISSUE THREAD

**Issue by pojntfx** [danielvladco/go-proto-gql#10]
In my simple example gRPC service (see [this repo](https://github.com/pojntfx/grpc-go-math)), I'm trying to make use of this library. However, after using it as specified with the following go generate instruction:

```bash
//go:generate bash -c "protoc --go_out=plugins=grpc,paths=source_relative:../ --gqlgencfg_out=paths=source_relative:../ --gql_out=svcdir=true,paths=source_relative:../ --gogqlgen_out=paths=source_relative,gogoimport=false:../ -I=../ -I=/usr/local/include -I=`go list -m -f \"{{.Dir}}\" github.com/danielvladco/go-proto-gql/pb` ../proto/math.proto"
```
I try to generate the files for `gqlgen`:

```bash
[pojntfx@thinkpadx1c3 grpc-go-math]$ tree lib/proto/
lib/proto/
└── math.proto

0 directories, 1 file
[pojntfx@thinkpadx1c3 grpc-go-math]$ go generate ./...
go: finding github.com/pojntfx/grpc-go-math/lib latest
go: finding github.com/pojntfx/grpc-go-math/lib/proto latest
[pojntfx@thinkpadx1c3 grpc-go-math]$ tree lib/proto/
lib/proto/
├── math.gqlgen.pb.go
├── math.gqlgen.pb.yml
├── math.pb.go
├── math.pb.graphqls
└── math.proto

0 directories, 5 files
[pojntfx@thinkpadx1c3 grpc-go-math]$ cd lib/proto/
[pojntfx@thinkpadx1c3 proto]$ go run github.com/99designs/gqlgen -v -c ./math.gqlgen.pb.yml 
[pojntfx@thinkpadx1c3 proto]$ tree .
.
├── exec.go
├── math.gqlgen.pb.go
├── math.gqlgen.pb.yml
├── math.pb.go
├── math.pb.graphqls
└── math.proto

0 directories, 6 files
```

As you can see, only the `exec.go` file gets generated; `models.go` etc. is missing. Here is the `math.gqlgen.pb.yml` (which looks fine to me):

```yaml
schema:
- math.pb.graphqls
exec:
  filename: exec.go
model:
  filename: models.go
models:
  Int64:
    model:
    - github.com/danielvladco/go-proto-gql/pb.Int64
  MathAddArgsInput:
    model:
    - github.com/pojntfx/grpc-go-math/lib/proto.MathAddArgs
  MathAddReply:
    model:
    - github.com/pojntfx/grpc-go-math/lib/proto.MathAddReply
  MathSubtractArgsInput:
    model:
    - github.com/pojntfx/grpc-go-math/lib/proto.MathSubtractArgs
  MathSubtractReply:
    model:
    - github.com/pojntfx/grpc-go-math/lib/proto.MathSubtractReply
```

The following proto is the basis:

```bash
syntax = "proto3";

package math;
import "gql.proto";
option go_package = "github.com/pojntfx/grpc-go-math/lib/proto";

// Service math adds and subtracts numbers
service Math {
  rpc Add(MathAddArgs) returns (MathAddReply) {
    option (gql.rpc_type) = QUERY;
  };
  rpc Subtract(MathSubtractArgs) returns (MathSubtractReply) {
    option (gql.rpc_type) = QUERY;
  };
}

// MathAddArgs are the arguments for an addition
message MathAddArgs {
  int64 First = 1;
  int64 Second = 2;
}

// MathAddReply is the reply for an addition
message MathAddReply { int64 Result = 3; }

// MathSubtractArgs are the arguments for a subtraction
message MathSubtractArgs {
  int64 First = 1;
  int64 Second = 2;
}

// MathSubtractReply is the reply for a subtraction
message MathSubtractReply { int64 Result = 3; }
```

The generated `.graphqls` looks fine to me as well:

```bash
# Code generated by protoc-gen-gogql. DO NOT EDIT
# source: proto/math.proto

directive @Math(name: String) on FIELD_DEFINITION

scalar Int64

input MathAddArgsInput {
	first: Int64  
	second: Int64  
}

input MathSubtractArgsInput {
	first: Int64  
	second: Int64  
}

type MathAddReply {
	result: Int64 
}

type MathSubtractReply {
	result: Int64 
}

type Query {
	mathAdd(in: MathAddArgsInput): MathAddReply @Math
	mathSubtract(in: MathSubtractArgsInput): MathSubtractReply @Math
}
```

To summarize, it appears as though only the `exec.go` file gets generated; this however is not sufficient enough to build an actual GraphQL server. I'm still quite new to Golang, so there might be something obvious wrong here; sorry if that is the case.

**Comment by danielvladco** [comment]
Hi @pojntfx , You don't actually need `models.go` because all the required models are already generated by `protoc-gen-go` plugin. 
The rest of the 'glue' code is generated with `protoc-gen-gogqlgen` so you actually don't need to generate a models.go file in order to build your server.

**Comment by pojntfx** [comment]
@danielvladco Thanks for the quick response! How exactly does one implement the server then in this case? The example in this repo does not have an example server implemented and the `gqlgen` docs work with the models file ...

**Comment by danielvladco** [comment]
The generated code in *.gqlgen.pb.go will have a structure like this
```go
type MathGQLServer struct{ Service MathServer }
```

So the easiest way to implement it is like this (assuming all the resolvers are of query type and there is only one service):

```go

type root struct {
	mathSvc       pb.MathServer
}

func (r *root) Query() gql.QueryResolver {
	return pb.MathGQLServer{Service: r.mathSvc}
}

```

And then you already know ;)

```go
http.Handle("/graphql", handler.GraphQL(gql.NewExecutableSchema(gql.Config{
		Resolvers: &root{mathSvc: mathSvc},
}))
```

**Comment by pojntfx** [comment]
@danielvladco Thank you so much! I'll try and implement it this afternoon. I'm dealing with the following error right now:

```bash
vagrant@thinkbox-2:~$ grpcurl -plaintext localhost:30000 describe                                                       grpc.reflection.v1alpha.ServerReflection is a service:
service ServerReflection {
  rpc ServerReflectionInfo ( stream .grpc.reflection.v1alpha.ServerReflectionRequest ) returns ( stream .grpc.reflection.v1alpha.ServerReflectionResponse );
}
Failed to resolve symbol "math.Math": Symbol not found: math.Math
caused by: File not found: gql.proto
```

Which I am not yet able to solve (the above `go generate` command works just fine, so I guess it can find the `gql.proto` file at build time, but when I try to use the reflection API or call the service it fails. If I uncomment the import of the `gql.proto` file, everything works fine:

```proto
vagrant@thinkbox-2:~/Documents/grpc-go-math$ cat ./lib/proto/math.proto                                                 syntax = "proto3";

package math;
// import "gql.proto";
option go_package = "github.com/pojntfx/grpc-go-math/lib/proto";

// Service math adds and subtracts numbers
service Math {
  rpc Add(MathAddArgs) returns (MathAddReply) {
    // option (gql.rpc_type) = QUERY;
  };
  rpc Subtract(MathSubtractArgs) returns (MathSubtractReply) {
    // option (gql.rpc_type) = QUERY;
  };
}

// MathAddArgs are the arguments for an addition
message MathAddArgs {
  int64 First = 1;
  int64 Second = 2;
}

// MathAddReply is the reply for an addition
message MathAddReply { int64 Result = 3; }

// MathSubtractArgs are the arguments for a subtraction
message MathSubtractArgs {
  int64 First = 1;
  int64 Second = 2;
}

// MathSubtractReply is the reply for a subtraction
message MathSubtractReply { int64 Result = 3; }
```

```bash
vagrant@thinkbox-2:~/Documents/grpc-go-math$ grpcurl -plaintext localhost:30000 describe                                grpc.reflection.v1alpha.ServerReflection is a service:
service ServerReflection {
  rpc ServerReflectionInfo ( stream .grpc.reflection.v1alpha.ServerReflectionRequest ) returns ( stream .grpc.reflection.v1alpha.ServerReflectionResponse );
}
math.Math is a service:
service Math {
  rpc Add ( .math.MathAddArgs ) returns ( .math.MathAddReply );
  rpc Subtract ( .math.MathSubtractArgs ) returns ( .math.MathSubtractReply );
}
```

Though this of course won't compile with `gqlgen` because there will be no queries.

**Comment by danielvladco** [comment]
The .grpc.reflection.v1alpha.ServerReflectionRequest doesn't have anything to do with this implementation. You probably did not configure your server correctly.

the problem is probably with `grpcurl -plaintext localhost:30000 describe`. As I know you have to enable grpc on your server for this to work.

Also if you want to be able to serve grpc as well as graphql you have to start listen for both servers in separate go routines

**Comment by danielvladco** [comment]
@pojntfx , if you no longer experience the issue may I close the it?

**Comment by pojntfx** [comment]
@danielvladco Sorry for not answering earlier. Yes, the issue has been solved - I had misconfigured my local `protoc` setup; simply putting the `protoc` binary into `/usr/local/bin` isn't enough, it also requires the `include` etc. things. Thanks for your help!
